components/style/properties/shorthand/box.mako.rs

```diff
@@ -308,28 +308,55 @@ macro_rules! try_parse_one {
 
     impl<'a> LonghandsToSerialize<'a>  {
         fn to_css_declared<W>(&self, dest: &mut W) -> fmt::Result where W: fmt::Write {
-            try!(self.animation_duration.to_css(dest));
-            try!(write!(dest, " "));
-
-            try!(self.animation_timing_function.to_css(dest));
-            try!(write!(dest, " "));
-
-            try!(self.animation_delay.to_css(dest));
-            try!(write!(dest, " "));
-
-            try!(self.animation_direction.to_css(dest));
-            try!(write!(dest, " "));
-
-            try!(self.animation_fill_mode.to_css(dest));
-            try!(write!(dest, " "));
+            fn extract_value<T>(x: &DeclaredValue<T>) -> Option< &T> {
+                match *x {
+                    DeclaredValue::Value(ref val) => Some(val),
+                    _ => None,
+                }
+            }
 
-            try!(self.animation_iteration_count.to_css(dest));
-            try!(write!(dest, " "));
+            use std::cmp;
+            let mut len = 0;
+            % for name in "duration timing_function delay direction fill_mode iteration_count play_state name".split():
```

Comment:
Done. Thanks @upsuper for the review and feedback - I'll take a stab at the other issue over the weekend.